### PR TITLE
[DDW-592] Stake pool list consistency

### PR DESCRIPTION
**Pull request by lucas-barros**
This PR adds table view for delegated stake pools list

## Todos

- [ ] TODO

## Screenshots

image
image

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1643053652128300)
- [ ] Test

---

## Test scenarios

**Scenario 1 - Validate tile view (by ranking)**
```gherkin
Given that I am on the stake pool screen
And the stake pools are displayed on tiles (by ranking)
And I have at least one Shelley wallet that is delegated
When I validate the screen
Then I can see all the stake pools I'm delegating at the moment at the top on a separate section
And these are displayed as tiles too, showing the corresponding ranking for each stake pool
```

**Scenario 2 - Validate table view**
```gherkin
Given that I am on the stake pool screen
And the stake pools are displayed on a table
And I have at least one Shelley wallet that is delegated
When I validate the screen
Then I can see all the stake pools I'm delegating at the moment at the top on a separate section
And these are displayed on a table too
And this list can be sorted by each header 
```

**Scenario 3 - Validate tile-rewards view (by potential rewards)**
```gherkin
Given that I am on the stake pool screen
And the stake pools are displayed on tiles (by potential rewards)
And I have at least one Shelley wallet that is delegated
When I validate the screen
Then I can see all the stake pools I'm delegating at the moment at the top on a separate section
And these are displayed as tiles too, showing the potential rewards for each stake pool
```

---

### Basics
- [x] PR assigned to the PR author(s) 
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [x] All visual regression testing has been reviewed (Go to Github Actions tab -> Select Chromatic workflow -> Run on your working branch)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] ~~Text changes are proofread and approved (Jane Wild / Amy Reeve)~~
- [x] ~~Japanese text changes are proofread and approved (Junko Oda)~~
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [x] ~~In case of dependency changes `yarn.lock` file is updated~~

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/staking/stake-pools/StakePools.js [hunk 1]
```diff
@@ -293,21 +293,43 @@ export default class StakePools extends Component<Props, State> {
...
-                      onSelect={this.onDelegate}
-                      numberOfRankedStakePools={numberOfRankedStakePools}
-                      isGridRewardsView={isGridRewardsView}
-                      showWithSelectButton
-                      scrollElementRef={
-                        stakePoolsScrollContext.scrollElementRef
-                      }
-                    />
-                  )}
+                  {(stakePoolsScrollContext) =>
```
Comment (mchappell) [inline source/renderer/app/components/staking/stake-pools/StakePools.js:296]: if the `stakePoolsList` component is the only one using `<StakingPageScrollContext.Consumer>` why not start off with the `isListView ? ` check and only include the consumer for `stakePoolsList` component

Comment (lucas-barros) [reply source/renderer/app/components/staking/stake-pools/StakePools.js]: Right, updated. Thanks!

**Comment by alexander-rukin** [comment]
image

It feels that table view now has a few problems:
- table header starts in scrolled view 
- the last element has a separator after 

References:  https://zpl.io/2jyD0Zq + https://zpl.io/bovpjWo

p.s. I feel like this one should be fixed separately -> pls create a proper ticket for this 🙏

**Review by alexander-rukin — APPROVED** [review]
looks consistent now

**Review by gabriela-ponce — APPROVED** [review]

**Review by ManusMcCole — APPROVED** [review]
Looks good @lucas-barros 👍

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Tested on 20611. Great work @lucas-barros 👍

**Pull request merged**